### PR TITLE
Implement refactorings (extract method/introduce local) suggested during code review.

### LINKED ISSUE #13842: Move code that handles Out Variable in Binder.BindArgumentValue into a separate method.
Suggested by @gafter at #13818.

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #13842.

@dotnet/roslyn-compiler, @jcouv, @gafter Please review.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
Ping. @dotnet/roslyn-compiler, @jcouv, @gafter Please review.

File: src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs [hunk 1]
```diff
@@ -20229,9 +20229,15 @@ public static bool TakeOutParam<T>(T y, out T x)
 
                 var tree = compilation.SyntaxTrees.Single();
                 Assert.Empty(GetOutVarDeclarations(tree));
+                AssertNoGlobalStatements(tree);
```
Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: >AssertNoGlobalStatements [](start = 16, length = 24)

Why doesn't
`H.TakeOutParam(1, out int x1);`
qualify as a global statement?

Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: Parser doesn't parse anything as a global statement, unless it is in a Script mode.

Comment (gafter) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs]: Oh, I see that you parsed the source twice, once in script mode and once without.

---
In reply to: [79447599](https://github.com/dotnet/roslyn/pull/13885#discussion_r79447599) [](ancestors = 79447599)

**Comment by gafter** [comment]
:shipit:

**Comment by jcouv** [comment]
:shipit:

**Pull request merged**